flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperatorFactory.java

```diff
@@ -61,16 +59,16 @@ public void setMailboxExecutor(MailboxExecutor mailboxExecutor) {
 	}
 
 	@Override
-	public StreamOperator createStreamOperator(StreamTask containingTask, StreamConfig config, Output output) {
+	public <T extends StreamOperator<OUT>> T createStreamOperator(StreamOperatorInitializer<OUT> initializer) {
```

Comment:
@rkhachatryan wanted to get rid of this method's generic afaik. It's really a bit anti pattern. So I'm not sure going into it makes any sense.